processor/awscontainerinsightprocessor/processor_test.go

```diff
@@ -0,0 +1,69 @@
+// Copyright  OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package awscontainerinsightprocessor
+
+import (
+	"os"
+	"sync"
+	"testing"
+	"time"
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+)
+
+type MockStore struct {
+	RefreshCount int
+	wg           sync.WaitGroup
+}
+
+func (m *MockStore) Decorate(metric pdata.ResourceMetrics, kubernetesBlob map[string]interface{}) bool {
+	return true
+}
+
+func (m *MockStore) RefreshTick() {
+	m.RefreshCount++
+	m.wg.Done()
+}
+
+func (m *MockStore) currentRefreshCount() int {
+	m.wg.Wait()
+	return m.RefreshCount
+}
+
+func TestStartAndShutdown(t *testing.T) {
+	os.Setenv("HOST_NAME", "host_name")
+	os.Setenv("HOST_IP", "host_ip")
```

Comment:
This modifies global state that could affect other tests.  This should store the current state of those environment variables and restore them at the end of the test.

Reply:
Sure. I will change it